lib/ui/pages/workspace_page/components/top_bar/top_app_bar.dart

```diff
@@ -1,16 +1,81 @@
 // Flutter imports:
 import 'package:flutter/material.dart';
-
+// Package imports:
+import 'package:flutter_riverpod/flutter_riverpod.dart';
 // Project imports:
+import 'package:paintroid/core/enums/tool_types.dart';
+import 'package:paintroid/core/providers/state/checkmark_clicked_state.dart';
+import 'package:paintroid/core/providers/state/tools/toolbox/toolbox_state_provider.dart';
+import 'package:paintroid/core/tools/line_tool/line_tool.dart';
+import 'package:paintroid/core/tools/tool.dart';
 import 'package:paintroid/ui/pages/workspace_page/components/top_bar/overflow_menu.dart';
+import 'package:paintroid/ui/shared/action_button.dart';
+import 'package:paintroid/ui/utils/top_bar_action_data.dart';
 
-// Project imports:
+class TopAppBar extends ConsumerWidget implements PreferredSizeWidget {
+  final String title;
+
+  const TopAppBar({super.key, required this.title});
+
+  List<Widget> getActions(Tool currentTool, WidgetRef ref) {
+    List<Widget> actions = [
+      if (currentTool is LineTool && currentTool.vertexStack.isNotEmpty) ...[
```

Comment:
Would it not be nice if we create a bool in currentTool for each tool type like: 
```
currentTool.isLineTool, currentTool.isBrushTool
```

Reply:
Would be nice indeed! Would it make sense to create a refactoring ticket for that as its not directly related to the linetool?